addons/account/tests/test_reconciliation_matching_rules.py

```diff
@@ -347,66 +387,377 @@ def test_matching_fields_match_label(self):
-            ], 'model': self.rule_1, 'partner': self.bank_line_2.partner_id},
-            self.cash_line_1.id: {'aml_ids': [self.invoice_line_4.id], 'model': self.rule_1, 'partner': self.cash_line_1.partner_id},
-        })
-        self.rule_1.match_total_amount_param = 100.0
-        self.bank_line_1.amount -= 10
+    @freeze_time('2019-01-01')
+    def test_zero_payment_tolerance_inbound(self):
+        rule = self._create_rule(line_ids=[{}])
+
+        invl = self._create_invoice_line(1000.0, self.partner_a, 'out_invoice', date='2019-01-01')
+
+        # Exact matching.
+        st_line = self._create_st_line(amount=1000.0)
+        self._check_statement_matching(
+            rule,
+            {st_line.id: {'aml_ids': invl.ids, 'model': rule, 'partner': st_line.partner_id}},
+            statements=st_line.statement_id,
+        )
+
+        # No matching because there is no tolerance.
+        st_line = self._create_st_line(amount=990.0)
+        self._check_statement_matching(
+            rule,
+            {st_line.id: {'aml_ids': []}},
+            statements=st_line.statement_id,
+        )
+
+        # The payment amount is higher than the invoice one.
+        st_line = self._create_st_line(amount=1010.0)
+        self._check_statement_matching(
+            rule,
+            {st_line.id: {'aml_ids': invl.ids, 'model': rule, 'partner': st_line.partner_id}},
+            statements=st_line.statement_id,
+        )
+
+    @freeze_time('2019-01-01')
+    def test_zero_payment_tolerance_outbound(self):
+        rule = self._create_rule(line_ids=[])
+
+        invl = self._create_invoice_line(1000.0, self.partner_a, 'in_invoice', date='2019-01-01')
+
+        # Exact matching.
+        st_line = self._create_st_line(amount=-1000.0)
+        self._check_statement_matching(
+            rule,
+            {st_line.id: {'aml_ids': invl.ids, 'model': rule, 'partner': st_line.partner_id}},
+            statements=st_line.statement_id,
+        )
+
+        # No matching because there is no tolerance.
+        st_line = self._create_st_line(amount=-990.0)
+        self._check_statement_matching(
+            rule,
+            {st_line.id: {'aml_ids': []}},
+            statements=st_line.statement_id,
+        )
+
+        # The payment amount is higher than the invoice one.
+        st_line = self._create_st_line(amount=-1010.0)
+        self._check_statement_matching(
+            rule,
+            {st_line.id: {'aml_ids': invl.ids, 'model': rule, 'partner': st_line.partner_id}},
+            statements=st_line.statement_id,
+        )
+
+    @freeze_time('2019-01-01')
+    def test_zero_payment_tolerance_inbound_auto_reconcile(self):
+        rule = self._create_rule(
+            auto_reconcile=True,
+            line_ids=[{}],
+        )
+
+        invl = self._create_invoice_line(1000.0, self.partner_a, 'out_invoice',  pay_reference='123456', date='2019-01-01')
+
+        # The payment amount is higher than the invoice one.
```

Comment:
Maybe we should check the case where the payment amount is lower here as well, for completeness ?